ql/src/java/org/apache/hadoop/hive/ql/exec/repl/AtlasDumpTask.java

```diff
@@ -129,6 +137,14 @@ public int execute() {
     }
   }
 
+  @NotNull
+  private void initializeReplLogger(AtlasReplInfo atlasReplInfo) {
+    if (this.replLogger == null){
+      this.replLogger = new AtlasDumpLogger(atlasReplInfo.getSrcDB(),
+              atlasReplInfo.getStagingDir().toString());
+    }
+  }
```

Comment:
the new field + this method causes a behaviour change: the `replLogger` will be retained between `execute()` calls - it becomes lazy-init kinda stuff

I don't know this part - but could that cause any problem?

to 100% retain the old behavour you could rename this method to `getReplLogger()` and return a new instance if the stored is null

Reply:
That is a tricky one. Back then, when I created that draft, I didn't know how the TaskFactory is working. 

The lazy initialisation is because of `testAtlasDumpMetrics` literally verifies if the task ran successfully by testing if the logger is called. In my opinion, it is not the best test but also, this ticket is not about refactoring tests but changing a mocking framework. 

Let me think about the possible solution. The original test modified the objects inner state. That is a very bad pattern, of course so that was the reason why I added logger as an extra parameter for the class. 

The tricky thing is it is expected testing the logger calls but also, according to the factory class, we want to re-initialise it at every execute call. 

I'm pretty sure there is a solution by utilising mockito as well.

let me dig a little bit deeper on that and also double check the other ...Task files as well. Sorry, after about 3 quarters, my memory faded a little bit. 

Thank you for spotting it.

Reply:
@kgyrtkirk : It seems I can redo the original behaviour with upgrading mockito from 3.x to 4.x. 

The sad thing is that 4.x release contains a huge clean up and they removed a lot of deprecated methods that we already use. I create a jira about upgrading mockito to 4.x and I will continue this ticket after that. 

Side note: 
I've tried to upgrade mockito only in hive-exec but it is a tricky one and actually, 3.x is automatically pulled from the parent pom (in our case, it is our root pom.xml so I cannot do it in a module-by-module bases. 
But also, it contains a lot of small changes and it will be way easier to review it if I do it separately.

Reply:
Mockito upgrade story: https://issues.apache.org/jira/browse/HIVE-27515